cli/src/ui.rs

```diff
@@ -340,6 +340,18 @@ impl Ui {
         self.stdout().flush()?;
         let mut buf = String::new();
         io::stdin().read_line(&mut buf)?;
+
+        if buf.ends_with('\n') {
+            let mut chars = buf.chars();
+            chars.next_back();
+            buf = chars.as_str().to_owned();
+        } else if buf.is_empty() {
+            return Err(io::Error::new(
+                io::ErrorKind::UnexpectedEof,
+                "Prompt cancelled by EOF",
+            ));
```

Comment:
Maybe this should be a `user_error()` instead?

Reply:
This method returns `io::Result<>`, so we'd need to change that?  I feel like that's out of scope; 'unexpectedEof' is very accurate here.